src/OpenSearch/Aws/SigningClientDecorator.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace OpenSearch\Aws;
+
+use Aws\Credentials\CredentialsInterface;
+use Aws\Signature\SignatureInterface;
+use Psr\Http\Client\ClientInterface;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+
+/**
+ * A decorator client that signs requests using the provided AWS credentials and signer.
+ */
+class SigningClientDecorator implements ClientInterface
+{
+    public function __construct(
+        protected ClientInterface $inner,
+        protected CredentialsInterface $credentials,
+        protected SignatureInterface $signer,
+    ) {
+        if (!class_exists(SignatureInterface::class)) {
```

Comment:
This is broken. `SignatureInterfaceè is an interface, not a class, so this would always throw an exception when using it.

and the case of the SDK not being available cannot be handled inside the constructor: if there interface does not exists, there is no way to pass valid arguments to the constructor.